### ISSUE TITLE
Build logs should not include timestamps

### ISSUE THREAD

**Issue by smarterclayton** [openshift/origin#1647]
```
$ osc build-logs ruby-sample-build-4
2015-04-08T18:46:23.663573652Z E0408 18:46:23.662384       1 cfg.go:50] /root/.dockercfg: stat /root/.dockercfg: no such file or directory
2015-04-08T18:46:23.673295305Z I0408 18:46:23.673228       1 sti.go:111] Building 172.30.17.189:5000/default/origin-ruby-sample
2015-04-08T18:46:23.939941363Z Cloning into '/tmp/sti290541991/upload/src'...
2015-04-08T18:47:01.485715788Z I0408 18:47:01.485307       1 sti.go:231] Successfully built 172.30.17.189:5000/default/origin-ruby-sample
2015-04-08T18:47:02.690560758Z E0408 18:47:02.690396       1 cfg.go:50] /root/.dockercfg: stat /root/.dockercfg: no such file or directory
2015-04-08T18:47:02.690560758Z I0408 18:47:02.690420       1 dockerutil.go:22] Pushing 172.30.17.189:5000/default/origin-ruby-sample ...
2015-04-08T18:47:23.347668131Z F0408 18:47:23.346955       1 builder.go:62] Build error: API error (500):  v1 ping attempt failed with error: Get http://172.30.17.189:5000/v1/_ping: dial tcp 172.30.17.189:5000: i/o timeout
```

Getting timestamps from two sources.  It's probably due to the glog prefix, which probably means our STI code needs to stop using glog.

**Comment by smarterclayton** [comment]
I want us to be doing something like this https://github.com/cockroachdb/cockroach/pull/963

The big one is being able to disable the prefix, everything else is nice to have.

**Comment by mfojtik** [comment]
@smarterclayton +1 for cockroach 

are they accepting patches? (mean if we want to add something in future, we will not be locked as we are now with glog)

**Comment by jhadvig** [comment]
I've created a card for this issue so we track it
https://trello.com/c/GsiBwruf/594-replace-glog-with-other-logging-library-so-we-can-strip-the-header-from-logs

**Comment by smarterclayton** [comment]
Dunno - i think they've changed direction.  Post 3.0 we need to settle our logging story, the card will be where we discuss.

**Comment by mfojtik** [comment]
@smarterclayton i experimented with this couple weeks ago: https://github.com/mfojtik/source-to-image/blob/new-logging/cmd/sti/main.go#L271

(this is a rework of sti logging to use cockroach... however I can't figure out how to disable the prefix..)

**Comment by soltysh** [comment]
Before actually starting to choosing/implementing something, I'd say we should create a card and write down all the requirements we have, on top of my head:
* ability to remove prefix (date, loglevel, etc)
* ability to log to stdout and stderr on demand

**Comment by smarterclayton** [comment]
If also want the ability to do structural separation for the CLI

**Comment by bparees** [comment]
@smarterclayton structural separation of what?

a requirement from me would be:  abiility to enable logging for specific packages instead of just globally.

**Comment by smarterclayton** [comment]
Fabiano did a great write up of our CLI reqs a few months back - search for structured CLI in origin.

glog supports package filtering

**Comment by bparees** [comment]
@smarterclayton hm, someone else told me it did not..is it just that we don't expose a way to set the package in our log arguments?

**Comment by smarterclayton** [comment]
Yeah, they have like 15 options so we only expose loglevel right now

----- Original Message -----

**Comment by soltysh** [comment]
* be able to redirect output from docker to it (eg. see #2953 where we redirect to stdout instead of to logger subsystem).

**Comment by smarterclayton** [comment]
I'm liking the philosophy in https://github.com/coreos/etcd/tree/master/Godeps/_workspace/src/github.com/coreos/pkg/capnslog

**Comment by bparees** [comment]
i just like the name.

**Comment by smarterclayton** [comment]
Agree

**Comment by bparees** [comment]
i'm not entirely sure why or when this changed, but we no longer get duplicated timestamps in s2i builds:

```
I1029 21:04:18.490470       1 sti.go:115] The value of ALLOWED_UIDS is [1-]
I1029 21:04:18.490652       1 cfg.go:44] Locating docker auth for image openshif
t/ruby-20-centos7@sha256:1b449d62d5c731b50f0a1466e1971104225314a9f4a023af4b5e417
530c628f5 and type PULL_DOCKERCFG_PATH
I1029 21:04:18.490728       1 cfg.go:80] Using Docker authentication configurati
on in '/root/.docker/config.json'
I1029 21:04:18.490800       1 cfg.go:52] Problem accessing /root/.docker/config.
json: stat /root/.docker/config.json: no such file or directory
```

so marking this fixed.

**Comment by smarterclayton** [comment]
Hold up partner - we want to remove the timestamps.  Kube logs allows you
to include them if you want, so it's redundant for us to print them.

On Oct 29, 2015, at 5:33 PM, Ben Parees <notifications@github.com> wrote:

i'm not entirely sure why or when this changed, but we no longer get
duplicated timestamps in s2i builds:

I1029 21:04:18.490470       1 sti.go:115] The value of ALLOWED_UIDS is [1-]
I1029 21:04:18.490652       1 cfg.go:44] Locating docker auth for image openshif
t/ruby-20-centos7@sha256:1b449d62d5c731b50f0a1466e1971104225314a9f4a023af4b5e417
530c628f5 and type PULL_DOCKERCFG_PATH
I1029 21:04:18.490728       1 cfg.go:80] Using Docker authentication configurati
on in '/root/.docker/config.json'
I1029 21:04:18.490800       1 cfg.go:52] Problem accessing /root/.docker/config.
json: stat /root/.docker/config.json: no such file or directory

so marking this fixed.

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/issues/1647#issuecomment-152334584>.

**Comment by bparees** [comment]
well technically the issue you opened is in fact fixed, but ok we can repurpose this one.

**Comment by kargakis** [comment]
It's ugly having timestamps twice and after #5396 lands, this will be an issue.

**Comment by bparees** [comment]
@kargakis no one is arguing that.  but the duplicate timestamps that this was originally opened for were due to something else which appears to be resolved.

that k8s is going to start prepending its own timestamps to pod logs is a new challenge to be addressed.

**Comment by smarterclayton** [comment]
It won't prepend by default.  It's merely optional.  If it is prepending by
default, that is a bug.

**Comment by bparees** [comment]
> It won't prepend by default.  It's merely optional.  If it is prepending by
default, that is a bug.

right, all the more reason I was trying to say it's a new problem (or going to be a new problem when it lands and people use it).

**Comment by jhadvig** [comment]
@bparees I guess this one is solved

**Comment by kargakis** [comment]
@jhadvig there is still https://github.com/openshift/origin/issues/5788

**Comment by smarterclayton** [comment]
Can we fix builds to cleanup output?  Still using glog time prefixes.

**Comment by bparees** [comment]
does glog give us a way to not print timestamps when logging?  if not, are you requesting we convert all the origin-sti-builder and origin-docker-builder logging statements to printlns?

**Comment by bparees** [comment]
(which is basically impossible since i'm sure there are is at least some shared library code that runs in both standalone and pod contexts)

**Comment by bparees** [comment]
btw this is what we currently do in s2i to allow us to avoid glog when printing the output of the assemble container:
https://github.com/openshift/source-to-image/blob/master/pkg/build/strategies/sti/sti.go#L545-L568

**Comment by smarterclayton** [comment]
Ok, so we never actually fixed this.  The original solution still stands
(wrapping glog vs not).

**Comment by bparees** [comment]
@smarterclayton this was fixed by your glog fakery, right?

**Comment by smarterclayton** [comment]
yeah
